### PR TITLE
chore: upgrade storybook 7.6.3 + use @storybook/nextjs

### PR DESCRIPTION
**Pull request by ThyMinimalDev**
## What does this PR do?

Upgrade storybook to latest version + use  @storybook/nextjs + fix breaking changes

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- run storybook in local

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/ui/components/form/color-picker/colorpicker.stories.mdx [hunk 1]
```diff
@@ -1,4 +1,7 @@
 import { TooltipProvider } from "@radix-ui/react-tooltip";
+import {
+  Tooltip,
```
Comment (ThyMinimalDev) [inline packages/ui/components/form/color-picker/colorpicker.stories.mdx:3]: import was missing

File: packages/ui/components/dialog/dialog.stories.mdx [hunk 2]
```diff
@@ -53,6 +53,11 @@ The `Dialog` component is composed of the following components:
 <Canvas>
   <Story
     name="Dialog"
+    parameters={{
+      nextjs: {
+        appDirectory: true,
+      },
+    }}
```
Comment (ThyMinimalDev) [inline packages/ui/components/dialog/dialog.stories.mdx:60]: specify that this component uses features from next.js 13+ imported from next/navigation

File: packages/ui/components/form/color-picker/colorpicker.test.tsx [hunk 3]
```diff
@@ -46,14 +46,9 @@ describe("Tests for ColorPicker component", () => {
...
-    await act(async () => {
-      userEvent.clear(colorInput);
-    });
+    await act(async () => userEvent.clear(colorInput));
     const newColorValue = "#00FF00";
-    await act(async () => {
-      userEvent.type(colorInput, newColorValue);
-    });
-
+    await act(async () => userEvent.type(colorInput, newColorValue));
```
Comment (ThyMinimalDev) [inline packages/ui/components/form/color-picker/colorpicker.test.tsx:51]: fix flaky test

**Review by emrysal — APPROVED** [review]
Looks good to me, nice improvement over what was.

**Pull request merged**